src/Func.h

```diff
@@ -1527,11 +1537,47 @@ class Func {
      * Hexagon, that loop is executed on a Hexagon DSP. */
     EXPORT Func &hexagon(VarOrRVar x = Var::outermost());
 
-    /** Prefetch data read by a subsequent loop iteration, at an
-     * optionally specified iteration offset. This is currently only
-     * implemented on Hexagon, prefetch directives are ignored on
-     * other targets. */
-    EXPORT Func &prefetch(VarOrRVar var, Expr offset = 1);
+    /** Prefetch data written to or read from a Func or an ImageParam by a
+     * subsequent loop iteration, at an optionally specified iteration offset.
+     * 'var' specifies at which loop level the prefetch calls should be inserted.
+     * The final argument specifies how prefetch of region outside bounds
+     * should be handled.
+     *
+     * For example, consider this pipeline:
+     \code
+     Func f, g;
+     Var x, y;
+     f(x, y) = x + y;
+     g(x, y) = 2 * f(x, y);
+     \endcode
+     *
+     * The following schedule:
+     \code
+     f.compute_root();
+     g.prefetch(f, x, 2);
+     \endcode
+     *
+     * will inject prefetch call at the innermost loop of 'g' and generate
+     * the following loop nest:
+     * for y = ...
+     *   for x = ...
+     *     f(x, y) = x + y
+     * for y = ..
+     *   for x = ...
+     *     prefetch(f, extent, stride);
```

Comment:
Clarify extent and stride here. In this case, wouldn't the extent be 1? Or would the prefetch even just be zero dimensional?

Also, it seems like the prefetch in this comment should indicate that it would be prefetch(&f[x + 2, y])